### ISSUE TITLE
Build fails with `pread failed: Is a directory`

### ISSUE THREAD

**Issue by baloe** [oneapi-src/oneTBB#560]
Hello,

I intending to build TBB on an HPC, but I'm stuck.

This is my installer SLURM job script:

```
#!/usr/bin/env bash

#SBATCH --cpus-per-task=12
#SBATCH --mem-per-cpu=2541
#SBATCH --nodes=1
#SBATCH --ntasks=1

#SBATCH --gres=gpu:1
#SBATCH --partition=gpu2

module load fosscuda/2020b # GCCcore-10.2.0
module load Python/3.8.6-GCCcore-10.2.0
module load SciPy-bundle/2020.11-fosscuda-2020b # < superbuild will build scipy anyway
module load matplotlib/3.3.3-fosscuda-2020b
module load CMake/3.18.4-GCCcore-10.2.0
module load libpng/1.6.37-GCCcore-10.2.0
module load Ninja/1.10.1-GCCcore-10.2.0
module load gperf/3.1-GCCcore-10.2.0       # < superbuild will build this anyway
module load libffi/3.3-GCCcore-10.2.0
module load libxml2/2.9.10-GCCcore-10.2.0
module load pybind11/2.6.0-GCCcore-10.2.0  # < superbuild will build this anyway
module load util-linux/2.36-GCCcore-10.2.0 # < superbuild will build this anyway
module load zlib/1.2.11-GCCcore-10.2.0
module load freetype/2.10.3-GCCcore-10.2.0
module load SQLite/3.33.0-GCCcore-10.2.0
module load fontconfig/2.13.92-GCCcore-10.2.0 # < superbuild will probably build this anyway

version=v2021.3.0

git clone https://github.com/oneapi-src/oneTBB.git oneTBB_src
cd oneTBB_src
git checkout $version
cd ..

mkdir "oneTBB_${version}"
cd "oneTBB_${version}"

cmake -DCMAKE_BUILD_TYPE=Release ../oneTBB_src

cmake --build .

ctest
```

and this is the output generated in configure and build phase:
```
-- The CXX compiler identification is GNU 10.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /sw/installed/GCCcore/10.2.0/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Checking for one of the modules 'hwloc'
-- The tbbbind_2_0 target will be configured using the HWLOC 2.2.0
-- The C compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /sw/installed/GCCcore/10.2.0/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done
WARNING: Target "conformance_arena_constraints" requests linking to directory "/sw/installed/hwloc/2.2.0-GCCcore-10.2.0/lib".  Targets may link only to libraries.  CMake is dropping the item.
WARNING: Target "test_arena_constraints" requests linking to directory "/sw/installed/hwloc/2.2.0-GCCcore-10.2.0/lib".  Targets may link only to libraries.  CMake is dropping the item.
-- Generating done
-- Build files have been written to: /scratch/ws/0/s4080143-soft/stage07/oneTBB_v2021.3.0
Scanning dependencies of target tbb
[  0%] Building CXX object src/tbb/CMakeFiles/tbb.dir/address_waiter.cpp.o
[  1%] Building CXX object src/tbb/CMakeFiles/tbb.dir/allocator.cpp.o
[  1%] Building CXX object src/tbb/CMakeFiles/tbb.dir/arena.cpp.o
[  1%] Building CXX object src/tbb/CMakeFiles/tbb.dir/arena_slot.cpp.o
[  2%] Building CXX object src/tbb/CMakeFiles/tbb.dir/concurrent_bounded_queue.cpp.o
[  2%] Building CXX object src/tbb/CMakeFiles/tbb.dir/dynamic_link.cpp.o
[  2%] Building CXX object src/tbb/CMakeFiles/tbb.dir/exception.cpp.o
[  3%] Building CXX object src/tbb/CMakeFiles/tbb.dir/governor.cpp.o
[  3%] Building CXX object src/tbb/CMakeFiles/tbb.dir/global_control.cpp.o
[  3%] Building CXX object src/tbb/CMakeFiles/tbb.dir/itt_notify.cpp.o
[  4%] Building CXX object src/tbb/CMakeFiles/tbb.dir/main.cpp.o
[  4%] Building CXX object src/tbb/CMakeFiles/tbb.dir/market.cpp.o
[  4%] Building CXX object src/tbb/CMakeFiles/tbb.dir/misc.cpp.o
[  5%] Building CXX object src/tbb/CMakeFiles/tbb.dir/misc_ex.cpp.o
[  5%] Building CXX object src/tbb/CMakeFiles/tbb.dir/observer_proxy.cpp.o
[  5%] Building CXX object src/tbb/CMakeFiles/tbb.dir/parallel_pipeline.cpp.o
[  6%] Building CXX object src/tbb/CMakeFiles/tbb.dir/private_server.cpp.o
[  6%] Building CXX object src/tbb/CMakeFiles/tbb.dir/profiling.cpp.o
[  6%] Building CXX object src/tbb/CMakeFiles/tbb.dir/rml_tbb.cpp.o
[  7%] Building CXX object src/tbb/CMakeFiles/tbb.dir/rtm_mutex.cpp.o
[  7%] Building CXX object src/tbb/CMakeFiles/tbb.dir/rtm_rw_mutex.cpp.o
[  7%] Building CXX object src/tbb/CMakeFiles/tbb.dir/semaphore.cpp.o
[  7%] Building CXX object src/tbb/CMakeFiles/tbb.dir/small_object_pool.cpp.o
[  8%] Building CXX object src/tbb/CMakeFiles/tbb.dir/task.cpp.o
[  8%] Building CXX object src/tbb/CMakeFiles/tbb.dir/task_dispatcher.cpp.o
[  8%] Building CXX object src/tbb/CMakeFiles/tbb.dir/task_group_context.cpp.o
[  9%] Building CXX object src/tbb/CMakeFiles/tbb.dir/version.cpp.o
[  9%] Building CXX object src/tbb/CMakeFiles/tbb.dir/queuing_rw_mutex.cpp.o
[  9%] Linking CXX shared library ../../gnu_10.2_cxx11_64_release/libtbb.so
[  9%] Built target tbb
Scanning dependencies of target tbbmalloc
[  9%] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/backend.cpp.o
[ 10%] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/backref.cpp.o
[ 10%] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/frontend.cpp.o
[ 10%] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/large_objects.cpp.o
[ 11%] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/tbbmalloc.cpp.o
[ 11%] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/__/tbb/itt_notify.cpp.o
[ 11%] Linking CXX shared library ../../gnu_10.2_cxx11_64_release/libtbbmalloc.so
[ 11%] Built target tbbmalloc
Scanning dependencies of target tbbmalloc_proxy
[ 12%] Building CXX object src/tbbmalloc_proxy/CMakeFiles/tbbmalloc_proxy.dir/function_replacement.cpp.o
[ 12%] Building CXX object src/tbbmalloc_proxy/CMakeFiles/tbbmalloc_proxy.dir/proxy.cpp.o
[ 12%] Linking CXX shared library ../../gnu_10.2_cxx11_64_release/libtbbmalloc_proxy.so
[ 12%] Built target tbbmalloc_proxy
Scanning dependencies of target tbbbind_2_0
[ 13%] Building CXX object src/tbbbind/CMakeFiles/tbbbind_2_0.dir/tbb_bind.cpp.o
[ 13%] Linking CXX shared library ../../gnu_10.2_cxx11_64_release/libtbbbind_2_0.so
/sw/installed/binutils/2.35-GCCcore-10.2.0/bin/ld.gold: fatal error: /sw/installed/hwloc/2.2.0-GCCcore-10.2.0/lib: pread failed: Is a directory
collect2: error: ld returned 1 exit status
gmake[2]: *** [gnu_10.2_cxx11_64_release/libtbbbind_2_0.so.3.3] Error 1
gmake[1]: *** [src/tbbbind/CMakeFiles/tbbbind_2_0.dir/all] Error 2
gmake: *** [all] Error 2
```

**Comment by ivankochin** [comment]
Hi @baloe,
Thanks for reporting!

The issue root cause is incorrect usage of the `pkg_config` CMake module during the [HWLOC detection](https://github.com/oneapi-src/oneTBB/blob/v2021.3.0/cmake/hwloc_detection.cmake#L49). This issue is already fixed in the 2021.4 release, and the 2021.4 release candidate is already available as a [tag](https://github.com/oneapi-src/oneTBB/tree/v2021.4.0-rc1) on GitHub.

But if you cannot switch to the 2021.4 release candidate or wait for the official 2021.4 release, you may just disable the TBBbind library build by defining the `TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH` CMake variable (read more [here](https://github.com/oneapi-src/oneTBB/blob/master/cmake/README.md#tbbbind-library-configuration)).

The absence of the TBBbind library won't affect the correctness but may affect the performance if you use `tbb::info` interfaces.

**Comment by anton-potapov** [comment]
@baloe, please reopen if problem still exists.
